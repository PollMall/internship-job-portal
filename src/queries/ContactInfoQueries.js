import { gql } from '@apollo/client';

export const UPDATE_CONTACT_INFO = gql`
mutation(
  $id: Int!
  $email: String
  $phone: String
  $city: String
  $countryId: Int
  $website: String
  $avatarUrl: String
  $about: String
) {
  updateContactInfo(
    id: $id
    email: $email
    phone: $phone
    city: $city
    countryId: $countryId
    website: $website
    avatarUrl: $avatarUrl
    about: $about
  ) {
    id
    email
    phone
    city
    country {
      id
      name
    }
    website
    avatarUrl
    about
  }
}
`;

export const CREATE_CONTACT_INFO = gql`
mutation(
  $email: String!
  $phone: String!
  $city: String!
  $countryId: Int!
  $website: String
  $avatarUrl: String
  $about: String
) {
  createContactInfo(
    email: $email
    phone: $phone
    city: $city
    countryId: $countryId
    website: $website
    avatarUrl: $avatarUrl
    about: $about
  ) {
    id
    email
    phone
    city
    country {
      id
      name
    }
    website
    avatarUrl
    about
  }
}
`;
