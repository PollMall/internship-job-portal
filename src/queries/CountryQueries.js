import { gql } from '@apollo/client';

export const GET_COUNTRIES = gql`
query {
  counties {
    id
    name
  }
}
`;
