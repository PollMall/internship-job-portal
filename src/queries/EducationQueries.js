import { gql } from '@apollo/client';

export const CREATE_USER_EDUCATION = gql`
mutation(
  $institution: String
  $description: String
  $userId: Int!
  $startDate: String
  $endDate: String
) {
  createUserEducation(
    institution: $institution
    description: $description
    userId: $userId
    startDate: $startDate
    endDate: $endDate
  ) {
    id
    institution
    description
    startDate
    endDate
  }
}
`;

export const DELETE_USER_EDUCATION = gql`
mutation($id: [Int!]!) {
  deleteUserEducation(id: $id)
}
`;

export const UPDATE_USER_EDUCATION = gql`
mutation(
  $id: Int!
  $institution: String
  $description: String
  $userId: Int!
  $startDate: String
  $endDate: String
) {
  updateUserEducation(
    id: $id
    institution: $institution
    description: $description
    userId: $userId
    startDate: $startDate
    endDate: $endDate
  ) {
    id
    institution
    description
    startDate
    endDate
  }
}
`;
