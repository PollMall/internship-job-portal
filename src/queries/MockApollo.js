/* eslint-disable import/prefer-default-export */

export const mockResponseType = {
  SUCCESS: 'SUCCESS',
  ERROR: 'ERROR',
};

export const makeMock = (mockData, type) => {
  const { query, variables, response } = mockData;

  switch (type) {
    case mockResponseType.SUCCESS:
      return {
        request: {
          query,
          variables,
        },
        result: {
          data: response,
        },
      };
    case mockResponseType.ERROR:
      return {
        request: {
          query,
          variables,
        },
        error: response,
      };
    default:
      return new Error('Mock response not supported');
  }
};
