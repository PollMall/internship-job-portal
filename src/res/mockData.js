export const users = [
  {
    id: 1,
    username: 'a',
    password: 'a',
    firstName: 'a',
    lastName: 'a',
    userRole: {
      id: 3,
      name: 'user',
    },
  },
  {
    id: 2,
    username: 'b',
    password: 'b',
    firstName: 'b',
    lastName: 'b',
    userRole: {
      id: 3,
      name: 'user',
    },
  },
  {
    id: 3,
    username: 'c',
    password: 'c',
    firstName: 'c',
    lastName: 'c',
    userRole: {
      id: 3,
      name: 'user',
    },
  },
];

export const admin = {
  id: 1,
  username: 'c',
  password: 'c',
  firstName: 'c',
  lastName: 'c',
  userRole: {
    id: 1,
    name: 'sys_admin',
  },
};

export const userRoles = [
  {
    id: 1,
    name: 'sys_admin',
  },
  {
    id: 2,
    name: 'company_user',
  },
  {
    id: 3,
    name: 'user',
  },
];

export const contactInfo = {
  id: 1,
  email: 'email',
  phone: '+41234567890',
  city: 'city',
  country: {
    id: 1,
    name: 'country',
  },
  website: 'website',
  avatarUrl: 'url',
  about: 'about',
};

export const user = {
  id: 1,
  username: 'username',
  firstName: 'firstName',
  lastName: 'lastName',
  password: 'password',
  contactInfo,
  userEducations: [
    {
      id: 1,
      institution: 'institution',
      description: 'description',
      startDate: '1538265600',
      endDate: '1617235200',
    },
  ],
  userWorkExperiences: [
    {
      id: 1,
      institution: 'institution',
      description: 'description',
      startDate: '1617148800',
      endDate: '1617235200',
    },
  ],
  userSkills: [
    {
      id: 1,
      skill: {
        id: 1,
        name: 'HTML',
      },
      rating: 5,
    },
    {
      id: 2,
      skill: {
        id: 2,
        name: 'CSS',
      },
      rating: 5,
    },
  ],
  updatedAt: 1617309323,
};
