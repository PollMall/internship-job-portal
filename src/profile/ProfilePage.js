import React, { useContext } from 'react';
import { useQuery } from '@apollo/client';
import { GET_USER } from '../queries/UserQueries';
import { UserContext } from '../UserProvider';
import Info from '../Info';
import UserProfile from './UserProfile';
import UserProfileProvider from './UserProfileProvider';

function ProfilePage() {
  const { state } = useContext(UserContext);
  const {
    data, loading, error,
  } = useQuery(GET_USER, { variables: { id: state.user?.id } });

  return (
    <>
      <Info loading={loading} error={!!error} />
      {data
      && (
      <UserProfileProvider initialValue={{ user: data?.user }}>
        <UserProfile />
      </UserProfileProvider>
      )}
    </>
  );
}

export default ProfilePage;
