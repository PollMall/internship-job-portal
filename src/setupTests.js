// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';
import { MemoryRouter } from 'react-router-dom';
import { MockedProvider } from '@apollo/client/testing';
import { render } from '@testing-library/react';
import { mockApolloData } from './queries/MockApolloData';

Object.defineProperty(window, 'matchMedia', {
  value: () => ({
    matches: false,
    addListener: () => {},
    removeListener: () => {},
  }),
});

Object.defineProperty(window, 'alert', {
  value: jest.fn()
})

function makeMockApolloCalls(props) {
  return props.map((p) => {
    const { query, variables, data} = p;
    console.log(data);
    return {
      request: {
        query,
        variables: variables || {}
      },
      result: {
        data: data || {}
      }
    }
  })
}

const mockApolloCalls = makeMockApolloCalls(mockApolloData);

const AllProviders = ({ children }) => {
  return (
    <MockedProvider mocks={mockApolloCalls} addTypename={false}>
       <MemoryRouter>
         {children}
       </MemoryRouter>
     </MockedProvider>
  )
}

const customRender = (ui, options) =>
  render(ui, { wrapper: AllProviders, ...options })


export { customRender as render };