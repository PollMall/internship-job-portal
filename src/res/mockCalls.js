import { users, userRoles } from './mockData';
import { GET_USERS, DELETE_USER, GET_USER_ROLES } from '../queries/UserQueries';

export const mockGetUsers = {
  query: GET_USERS,
  variables: {},
  response: {
    users,
  },
};

export const mockDeleteUser = {
  query: DELETE_USER,
  variables: { id: 1 },
  response: {
    deleteUser: true,
  },
};

export const getUserRoles = {
  query: GET_USER_ROLES,
  variables: {},
  response: {
    userRoles,
  },
};
