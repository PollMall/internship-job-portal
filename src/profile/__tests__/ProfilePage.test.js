import React from 'react';
import { screen, waitFor } from '@testing-library/react';
import ProfilePage from '../ProfilePage';
import { render } from '../../setupTests';
import { getUser } from '../../res/mockCalls';
import { users } from '../../res/mockData';
import { makeMock, mockResponseType } from '../../queries/MockApollo';

describe('ProfilePage tests', () => {
  it('should match snapshot', () => {
    const { container } = render(<ProfilePage />);
    expect(container).toMatchSnapshot();
  });

  it('should display user profile', async () => {
    render(<ProfilePage />, [makeMock(getUser, mockResponseType.SUCCESS)], { user: users[0] });
    await waitFor(() => {
      expect(screen.queryByTestId('info-loading')).not.toBeInTheDocument();
    });
  });
});
