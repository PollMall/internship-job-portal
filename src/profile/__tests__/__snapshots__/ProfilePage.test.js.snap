// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePage tests should match snapshot 1`] = `
<div>
  <div
    class="MuiCircularProgress-root makeStyles-progress-1 MuiCircularProgress-indeterminate"
    data-testid="info-loading"
    role="progressbar"
    style="width: 40px; height: 40px;"
  >
    <svg
      class="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx="44"
        cy="44"
        fill="none"
        r="20.2"
        stroke-width="3.6"
      />
    </svg>
  </div>
</div>
`;
